modules/daicard/src/utils/migrate.js

```diff
@@ -0,0 +1,51 @@
+import * as Connext from "connext";
+import { Zero } from "ethers/constants";
+import { formatEther } from "ethers/utils";
+
+import { toBN } from './bn';
+
+export const migrate = async (hubUrl, wallet, ethUrl, setMigrating) => {
+  console.log(`____________________Migration Started | hubUrl: ${hubUrl}`)
+  if (!hubUrl) { return; }
+  const legacy = await Connext.createClient({ ethUrl, hubUrl, mnemonic: wallet.mnemonic });
+  await legacy.start();
+  const state = (await legacy.getState());
+  const latestState = state.persistent.latestValidState
+
+  const tokenBalance = toBN(latestState.balanceTokenUser);
+  const etherBalance = toBN(latestState.balanceWeiUser);
+  console.log(`Legacy channel has a balance of $${formatEther(tokenBalance)}`);
+
+  const amountToken = tokenBalance.add(state.persistent.custodialBalance.balanceToken);
+  const amountWei = etherBalance.add(state.persistent.custodialBalance.balanceWei);
+
+  if (amountToken.gt(Zero) || amountWei.gt(Zero)) {
+    await setMigrating(true);
+
+    const withdrawalParams = {
+      exchangeRate: state.runtime.exchangeRate.rates.DAI,
+      tokensToSell: amountToken.toString(),
+      withdrawalWeiUser: amountWei.toString(),
+      weiToSell: "0",
+      withdrawalTokenUser: "0"
+    }
+
+    console.log(`Cashing out legacy channel: ${JSON.stringify(
+      withdrawalParams,
+      (key, value) => value._hex ? toBN(value._hex).toString() : value,
+      2,
+    )}`);
+
+    try {
+      // To debug: simulate a withdrawal by just waiting for a bit
+      await new Promise((res, rej) => setTimeout(res, 5000));
+      // await legacy.withdraw(withdrawalParams);
```

Comment:
im not clear on what this is simulating or how it empties the channel? seems like this would have to be done for the migrations

Reply:
what happens if someone refreshes *during* a migration, would it just freeze up and say that it's already migrating but never migrate?